pkg/controller/storagecluster/reconcile.go

```diff
@@ -958,3 +966,61 @@ fi`
 
 	return job
 }
+
+// inhibitPDBAlert patches the Alertmanager config to inhibit PDB alerts for Rook OSDs
+// Inhibition Rules will be added if there is at least one Storage Cluster and
+// be removed otherwise
+func inhibitPDBAlert(client client.Client) error {
+	storageClusterList := &ocsv1.StorageClusterList{}
+	err := client.List(context.TODO(), storageClusterList)
+	if err != nil {
+		return fmt.Errorf("couldn't list Storage Clusters. %v", err)
+	}
+	storageClusterLength := len(storageClusterList.Items)
+
+	alertmanagerConfigSecret := &corev1.Secret{}
+	alertmanagerConfigSecretName := "alertmanager-main"
+	alertmanagerConfigSecretNamespace := "openshift-monitoring"
+	err = client.Get(context.TODO(), types.NamespacedName{Name: alertmanagerConfigSecretName, Namespace: alertmanagerConfigSecretNamespace}, alertmanagerConfigSecret)
+	if err != nil {
+		return fmt.Errorf("couldn't fetch alertmanager config secret. %v", err)
+	}
+	newAlertmanagerConfigSecret := getInhibitionRules(alertmanagerConfigSecret.DeepCopy(), storageClusterLength)
+	err = client.Update(context.TODO(), newAlertmanagerConfigSecret)
+	if err != nil {
+		return fmt.Errorf("couldn't update alertmanager config secret. %v", err)
+	}
+	return nil
+}
+
+func getInhibitionRules(alertmanagerConfigSecret *corev1.Secret, storageClusterLength int) *corev1.Secret {
+	// DO NOT EDIT THE FORMATTING OF THIS STRING
+	// Whitespaces are important as it is directly converted into YAML
+	inhibitRules := `
+	- "equal":
+	  - "namespace"
+	  - "alertname"
+	  "source_match_re":
+	   "poddisruptionbudget" : "^rook-ceph-osd-[0-9]+$"
+	  "target_match_re":
+	    "poddisruptionbudget" : "^rook-ceph-osd-[0-9]+$"`
+	oldConfig := alertmanagerConfigSecret.Data["alertmanager.yaml"]
+	oldconfigBytes := bytes.SplitAfter(oldConfig, []byte("\"inhibit_rules\":"))
```

Comment:
Are we always guaranteed that the "inhibit_rules" field will be the last in the YAML manifest?

Reply:
No, in fact it will never be the last.
We split the file wherever we find `inhibit_rules:` and append rules to it.

Reply:
But when you do the split, you seem to be assuming that everything after `inhibit_rules:` are subsections of that block. So I ask again, is that lock always guaranteed to be last in the file?

Reply:
Yes. That's ensured by openshift-monitoring operator. There'll always be a block of default `inhibit_rules:`.

Reply:
As long as it's also guaranteed to be at the end of the manifest, that's fine by me.

Reply:
The user is free to edit/change the secret to dispatch notifications wherever needed. The default configuration secret is just a minimal configuration to get started. I'm not sure it's reasonable to have an operator modifying it under the hood.
At the very least, you need to parse the YAML data instead of manipulating text strings.

Reply:
@simonpasquier When I looked into the alertmanager job in cluster monitoring operator, it looked like it creates/updates the secret. But, from what you are saying, it creates the secret and doesn't update it (user has to do it). That definitely adds a few more cases to be handled.

>  I'm not sure it's reasonable to have an operator modifying it under the hood.

We can't ask the user to do it because it needs to be done on all clusters and that's an overhead we don't want to add.
We'd be happy getting rid of the PDB alert altogether, but as discussed elsewhere it has importance in some other projects.

> At the very least, you need to parse the YAML data instead of manipulating text strings.

I tried doing that but didn't find any schema or type definition for the config file. It's defined as a block of string and so I chose string manipulation (which is close to how we'd do it manually via CLI or Edit YAML options).
Is there an example that parses this string?